### ISSUE TITLE
PINNs Examples: Not Producing Correct Solution

### ISSUE THREAD

**Issue by thivin-zenteiq-ai** [zenoxml/SciREX#62]
While running the examples, the PINNs code seems to minimise both the PDE loss and the Boundary loss.. However, upon completion of training the Solution Learnt by the Neural Network is not the correct Solution

``` Epoch: 0, Loss: 1.5463e+03, L2 Error: 5.3262e-01
  5%|████▊                                                                                               | 970/20000 [00:03<00:53, 357.20it/s]Epoch: 1000, Loss: 1.4953e+01, L2 Error: 1.6334e+00
 10%|█████████▊                                                                                         | 1980/20000 [00:06<00:48, 371.12it/s]Epoch: 2000, Loss: 2.3435e+00, L2 Error: 1.3601e+00
 15%|██████████████▋                                                                                    | 2967/20000 [00:08<00:45, 372.35it/s]Epoch: 3000, Loss: 4.7902e-01, L2 Error: 1.4034e+00
 20%|███████████████████▋                                                                               | 3973/20000 [00:11<00:42, 381.47it/s]Epoch: 4000, Loss: 2.0641e-01, L2 Error: 1.4291e+00
 25%|████████████████████████▋                                                                          | 4987/20000 [00:14<00:39, 384.46it/s]Epoch: 5000, Loss: 1.2801e-01, L2 Error: 1.4356e+00
 28%|███████████████████████████▉                                                                       | 5653/20000 [00:15<00:37, 385.56it/s```

**Comment by thivin-zenteiq-ai** [comment]
The way in which gradients are current computed for PINNs ( Esspecially for second derivative ), was the main issue. 

``` with tf.GradientTape(persistent=True) as tape2:
                    tape2.watch(self.input_tensor)
                    # Compute the predicted values from the model
                    # Compute the predicted values from the model
                    predicted_values = self(self.input_tensor)

                # compute the gradients of the predicted values wrt the input which is (x, y)
                gradients = tape2.gradient(predicted_values, self.input_tensor)
                pred_grad_x = gradients[:, 0]
                pred_grad_y = gradients[:, 1]
                ```
                
                
 Based on Existing PINNs codes available online, its advised to either split the inputs and concatenate them before passing to the input layers to get the appropriate gradients or extact the components of the gradients first before passing them into another tape to calculate higher order derivatives. (Something like this)
 
 ``` with tf.GradientTape(persistent=True) as tape1:
                tape1.watch(self.input_tensor)

                predicted_values = self(self.input_tensor, training=True)
                    
                gradients = tape1.gradient(predicted_values, self.input_tensor)

                grad_x = gradients[:, 0:1]
                grad_y = gradients[:, 1:2]

            # Compute Laplacian
            pred_grad_xx = tape1.gradient(grad_x, self.input_tensor)[:, 0:1]
            pred_grad_yy = tape1.gradient(grad_y, self.input_tensor)[:, 1:2]
 ```
            
 The issue is fixed on [PR 63](https://github.com/zenoxml/SciREX/pull/63)
